packages/cdktf-cli/test/lib/cdktf-project.test.ts

```diff
@@ -350,6 +368,66 @@ describe("CdktfProject", () => {
       ]);
     });
 
+    it("error in an deploying stack does not abort already running stacks", async () => {
+      const events: any[] = [];
+      const cdktfProject = new CdktfProject({
+        synthCommand: "npx ts-node ./main.ts",
+        ...installFixturesInWorkingDirectory(
+          inNewWorkingDirectory(),
+          "parallel-error"
+        ),
+        onUpdate: (event) => {
+          events.push(event);
+        },
+      });
+
+      try {
+        await cdktfProject.deploy({
+          stackNames: ["stack1", "stack2", "stack3"],
+          parallelism: 100,
+          autoApprove: true,
+        });
+        throw new Error("This error should not be thrown");
+      } catch (e) {
+        expect(e).toMatchInlineSnapshot(`[Error: non-zero exit code 1]`);
+      }
+
+      const relevantEvents = events
+        .filter((e) => !e.type.includes("update"))
+        .map((e) => `${e.stackName || "global"}: ${e.type}`);
+
+      // the first 5 events have a stable order
```

Comment:
Do we even need these?

Reply:
I think it still makes sense to test that these events correctly occur. This helps spotting issues that cause the test to fail, that might not be directly related to the other events which we test. E.g. if one of those first five events was missing, the other assertions would fail and it would be hard to spot why everything shifted in the array by one element (due to the missing event that is never mentioned in the test).

Reply:
Fair point 👍